src/Orleans.Serialization/Buffers/Adaptors/PooledArrayBufferWriter.cs

```diff
@@ -190,65 +190,18 @@ private void Commit()
         _currentPosition = 0;
     }
 
-    private sealed class SequenceSegmentPool
```

Comment:
I have a branch where I largely rewrote this. I am sure it's in a non-working state now, though.
The rewrite was to support efficient message-oriented networking. I may dig it up